cmake/helpers.cmake

```diff
@@ -40,6 +40,54 @@ function(add_cppformat name)
     add_dependencies(cppformat cppformat-${name})
 endfunction()
 
+macro(set_ur_targets_list)
+    set(UR_TARGETS
+        collector
+        disjoint_pool
+        hello_world
+        test-context
+        test-device
+        test-event
+        test-loader-platforms
+        test-memory
+        test-params
+        test-platform
+        test-queue
+        test-sampler
+        test-usm
+        test-virtual_memory
+        umf_test-base
+        umf_test-disjointPool
+        umf_test-disjointPoolConfigParser
+        umf_test-memoryPool
+        umf_test-memoryProvider
+        umf_test_helpers
+        ur_adapter_null
+        ur_collector
+        ur_loader
+        usm_test-usmPoolManager
+        validation_test-leaks
+        validation_test-leaks_mt
+        validation_test-parameters
+    )
+endmacro()
```

Comment:
This list is going to be a pain to maintain.

Alternatively, why not wrap `add_executable`/`add_library` to set the UR specific options? This is a pattern used by LLVM and a number of other projects.